### DISCUSSION TITLE
Allow autofit dpi without word wrapping

### DISCUSSION THREAD

**Discussion by Bytologic** [libvips/libvips#2846] [Q&A] [answered]
Hi,

I'm using vips_text while giving the width and height without the dpi to allow for autofitting the dpi.

This will give me the following result:
![Wrap](https://user-images.githubusercontent.com/23726981/172375776-fcbed9ae-0ac4-448e-ada6-f895f4a920df.png)

However, I am looking for a result like this:
![NonWrap](https://user-images.githubusercontent.com/23726981/172375899-5125e6fd-b34d-4742-b4d6-3ef0ab0fe422.png)

The easiest option for me would be if there was a bool parameter that would enable or disable word wrapping. However, I understand that this would break older versions if they were to update to this new version. Instead my idea would be to add an optional argument to disable word wrapping if I add a width, so that it functions purely as a width for the box to look at dpi's instead of a word wrapping width. Would something like this be possible?

**Comment by jcupitt** [comment, marked as answer]
Hi @Bytologic,

You can use a non-breaking space character. Pango doesn't support `&nbsp;`, unfortunately, but you can insert the utf8 character yourself, eg.:

```
$ vips text x.png "hello world" --width 100 --height 200
```

Generates:

![x](https://user-images.githubusercontent.com/580843/172406367-de6e8261-55d1-4678-9065-d410ca4d8af6.png)

**Reply by Bytologic** [reply]
Works perfectly, thank you for your quick reply!

**Comment by jcupitt** [comment]
8.14 has a new `--wrap none` option to `text` to disable line breaks, see https://github.com/libvips/libvips/pull/3130
